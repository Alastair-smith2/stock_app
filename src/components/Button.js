import React, { Component } from "react";
import { View, Text, TouchableHighlight, StyleSheet } from "react-native";

export default class Button extends Component {
  constructor(props) {
    super(props);

    this.onPress = this.onPress.bind(this);
  }
  onPress = () => {
    this.props.buttonPress();
  };

  render() {
    return (
      <TouchableHighlight
        onPress={this.onPress}
        style={[styles.container, this.props.buttonStyle]}
      >
        <Text style={styles.label}>{this.props.title}</Text>
      </TouchableHighlight>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1
  },
  label: {
    fontSize: 20,
    color: "#fff"
  }
});
