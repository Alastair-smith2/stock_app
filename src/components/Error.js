import React, { Component } from "react";
import {
  View,
  Text,
  TextInput,
  StyleSheet,
  TouchableHighlight
} from "react-native";
import Instructions from "./Instructions";
import Search from "./Search";
import Button from "./Button";

export default class Error extends Component {
  constructor(props) {
    super(props);

    this.searchSubmit = this.searchSubmit.bind(this);
    this.onTextChange = this.onTextChange.bind(this);
    this.restart = this.restart.bind(this);
  }
  onTextChange = (type, props) => {
    console.log(type, props, "What are the props being passed through here?");
    this.props.onTextSearch(type, props);
  };

  searchSubmit = () => {
    this.props.submitSearch();
  };

  restart = () => {
    this.props.startAgain();
  };

  render() {
    return (
      <View style={styles.container}>
        <View style={styles.errorContainer}>
          <Text style={styles.error}>There was an error loading the data</Text>
        </View>
        <Instructions
          instructionContainer={styles.instructionContainer}
          label={this.props.instructionLabel}
        />
        <View style={styles.searchContainer}>
          {this.props.errorType == "currency" ? (
            <Search
              error={true}
              searchType={"currency"}
              currencyOne={this.props.currencyOne}
              typeOne="currencyOne"
              labelOne={this.props.currencyOneLabel}
              currencyTwo={this.props.currencyTwo}
              typeTwo="currencyTwo"
              labelTwo={this.props.currencyOneLabel}
              submitSearch={this.searchSubmit}
              onTextSearch={this.onTextChange}
              restart={this.restart}
            />
          ) : (
            <Search
              error={true}
              label={this.props.searchLabel}
              searchType={"stock"}
              value={this.props.search}
              type={"search"}
              submitSearch={this.searchSubmit}
              onTextSearch={this.onTextChange}
              restart={this.restart}
            />
          )}
        </View>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: "#fff"
  },
  errorContainer: {
    flex: 0.1,
    justifyContent: "center",
    alignItems: "center"
  },
  instructionContainer: {
    flex: 0.2,
    justifyContent: "center",
    alignItems: "center"
  },
  searchContainer: {
    flex: 0.3,
    marginLeft: 10,
    marginRight: 10
  }
});
