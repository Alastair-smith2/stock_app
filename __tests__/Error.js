import React from "react";
import { shallow } from "enzyme";

import Error from "../src/components/Error.js";
import Button from "../src/components/Button.js";

describe("The Error component", () => {
  let wrapper;
  let mockSubmit = jest.fn();
  let mockTextHandle = jest.fn();
  let mockStartAgain = jest.fn();
  beforeEach(() => {
    wrapper = shallow(
      <Error
        instructionLabel={"Enter a stock symbol"}
        errorType={"stock"}
        search={""}
        onTextSearch={mockTextHandle}
        submitSearch={mockSubmit}
        startAgain={mockStartAgain}
      />
    );
    wrapper.setProps({ errorType: "stock" });
  });
  describe("It should render", () => {
    it("Should render", () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe("Features", () => {
    it("Should render an Instruction item", () => {
      expect(wrapper.find("Instructions")).toHaveLength(1);
    });

    it("Should render a Search item", () => {
      expect(wrapper.find("Search")).toHaveLength(1);
      expect(wrapper.find("Search").props().searchType).toEqual("stock");
      wrapper.setProps({ errorType: "currency" });
      expect(wrapper.find("Search").props().searchType).toEqual("currency");
    });
  });

  describe("Functionality", () => {
    it("Should be able to handle a text change", () => {
      wrapper.instance().onTextChange("currencyOne", "GBP");
      expect(mockTextHandle.mock.calls.length).toEqual(1);
      expect(mockTextHandle.mock.calls[0][0]).toEqual("currencyOne");
    });

    it("Should be able to handle a button submit", () => {
      wrapper
        .find("Search")
        .props()
        .submitSearch();
      expect(mockSubmit.mock.calls.length).toEqual(1);
    });

    it("Should be able to handle a restart press", () => {
      wrapper
        .find("Search")
        .props()
        .restart();
      expect(mockStartAgain.mock.calls.length).toEqual(1);
    });
  });
});
