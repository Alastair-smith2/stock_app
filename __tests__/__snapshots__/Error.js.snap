// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Error component It should render Should render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Error
    errorType="stock"
    instructionLabel="Enter a stock symbol"
    onTextSearch={[MockFunction]}
    search=""
    startAgain={[MockFunction]}
    submitSearch={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 0.1,
              "justifyContent": "center",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "red",
              }
            }
          >
            There was an error loading the data
          </Text>
        </View>,
        <Instructions
          instructionContainer={
            Object {
              "alignItems": "center",
              "flex": 0.2,
              "justifyContent": "center",
            }
          }
          label="Enter a stock symbol"
        />,
        <View
          style={
            Object {
              "flex": 0.3,
              "marginLeft": 10,
              "marginRight": 10,
            }
          }
        >
          <Search
            error={true}
            label={undefined}
            onTextSearch={[Function]}
            placeholder="Please enter a stock code"
            restart={[Function]}
            searchType="stock"
            submitSearch={[Function]}
            type="search"
            value=""
          />
        </View>,
      ],
      "style": Object {
        "backgroundColor": "#fff",
        "flex": 1,
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "red",
              }
            }
          >
            There was an error loading the data
          </Text>,
          "style": Object {
            "alignItems": "center",
            "flex": 0.1,
            "justifyContent": "center",
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "accessible": true,
            "allowFontScaling": true,
            "children": "There was an error loading the data",
            "ellipsizeMode": "tail",
            "style": Object {
              "color": "red",
            },
          },
          "ref": null,
          "rendered": "There was an error loading the data",
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "instructionContainer": Object {
            "alignItems": "center",
            "flex": 0.2,
            "justifyContent": "center",
          },
          "label": "Enter a stock symbol",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Search
            error={true}
            label={undefined}
            onTextSearch={[Function]}
            placeholder="Please enter a stock code"
            restart={[Function]}
            searchType="stock"
            submitSearch={[Function]}
            type="search"
            value=""
          />,
          "style": Object {
            "flex": 0.3,
            "marginLeft": 10,
            "marginRight": 10,
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "error": true,
            "label": undefined,
            "onTextSearch": [Function],
            "placeholder": "Please enter a stock code",
            "restart": [Function],
            "searchType": "stock",
            "submitSearch": [Function],
            "type": "search",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 0.1,
                "justifyContent": "center",
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "red",
                }
              }
            >
              There was an error loading the data
            </Text>
          </View>,
          <Instructions
            instructionContainer={
              Object {
                "alignItems": "center",
                "flex": 0.2,
                "justifyContent": "center",
              }
            }
            label="Enter a stock symbol"
          />,
          <View
            style={
              Object {
                "flex": 0.3,
                "marginLeft": 10,
                "marginRight": 10,
              }
            }
          >
            <Search
              error={true}
              label={undefined}
              onTextSearch={[Function]}
              placeholder="Please enter a stock code"
              restart={[Function]}
              searchType="stock"
              submitSearch={[Function]}
              type="search"
              value=""
            />
          </View>,
        ],
        "style": Object {
          "backgroundColor": "#fff",
          "flex": 1,
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "red",
                }
              }
            >
              There was an error loading the data
            </Text>,
            "style": Object {
              "alignItems": "center",
              "flex": 0.1,
              "justifyContent": "center",
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "accessible": true,
              "allowFontScaling": true,
              "children": "There was an error loading the data",
              "ellipsizeMode": "tail",
              "style": Object {
                "color": "red",
              },
            },
            "ref": null,
            "rendered": "There was an error loading the data",
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "instructionContainer": Object {
              "alignItems": "center",
              "flex": 0.2,
              "justifyContent": "center",
            },
            "label": "Enter a stock symbol",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Search
              error={true}
              label={undefined}
              onTextSearch={[Function]}
              placeholder="Please enter a stock code"
              restart={[Function]}
              searchType="stock"
              submitSearch={[Function]}
              type="search"
              value=""
            />,
            "style": Object {
              "flex": 0.3,
              "marginLeft": 10,
              "marginRight": 10,
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "error": true,
              "label": undefined,
              "onTextSearch": [Function],
              "placeholder": "Please enter a stock code",
              "restart": [Function],
              "searchType": "stock",
              "submitSearch": [Function],
              "type": "search",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
